src/main/scala/chiseltest/experimental/Observe.scala

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiseltest.experimental
+
+import chisel3._
+import chisel3.util.experimental._
+import chisel3.experimental.IO
+
+object observe {
+  def apply[T <: Data](signal: T): T = {
+    val wire = WireInit(0.U.asTypeOf(chiselTypeOf(signal)))
+    BoringUtils.bore(signal, Seq(wire))
+    wire
+  }
+}
+
+object expose {
+
+  /** The method `expose` allows one to bring to a wrapped module, signals from instantiated sub-modules so
+    * they can be tested by peek/poke chiseltest. This avoid cluttering the original module with signals that
+    * would be used only by the test harness.
+    *
+    * Usage:
+    *
+    * {{{
+    * import chisel3._
+    *
+    * // This is the module and submodule to be tested
+    * class SubModule extends Module {
+    *   val reg  = Reg(UInt(6.W))
+    *   reg := 42.U
+    * }
+    *
+    * // Top module which instantiates the submodule
+    * class Top extends Module {
+    *   val submodule = Module(new SubModule)
+    * }
+    *
+    * // Generate the main project Verilog
+    * object Proj extends App {
+    *  (new chisel3.stage.ChiselStage).emitVerilog(
+    *    new Top()
+    *  )}
+    * }}}
+    *
+    * Then in your spec, test the `Top` module with `chiseltest` and `scalatest` while being
+    *      able to observe the submodule:
```

Comment:
should be something like: "while exposing the signal from the submodule to the testbench"